src/vllm_tgis_adapter/grpc/grpc_server.py

```diff
@@ -283,6 +287,8 @@ async def Generate(
             #     # Abort the request if the client disconnects.
             #     await self.engine.abort(f"{request_id}-{i}")
             #     return self.create_error_response("Client disconnected")
+            if res.prompt is None:
+                res.prompt = request_prompt_map[res.request_id]
```

Comment:
I think we can remove the `request_prompt_map` and just use the index of the request as our id:
```suggestion
                res.prompt = request.requests[i].prompt
```

Reply:
can it just be list rather than dict?

Reply:
Actually, I do think we want the additional list/dict here, but it should contain the truncated prompt.

Reply:
Or maybe not... using the `/completions` endpoint with vLLM with `echo: "true"` seems to also return the un-truncated text in its output.

Reply:
@tjohnson31415 I thought about doing like that, are we 100% sure that the order of the requests and the responses will line up each time? The map made sure we got the exact text based on the response